client/components/CountdownTimer.tsx

```diff
@@ -0,0 +1,81 @@
+import dayjs from 'dayjs';
+import React from 'react';
+import { View, Text, StyleSheet } from 'react-native';
+
+import { useCountdown } from '../services/countdown';
+import DateTimeDisplay from './DateTimeDisplay';
+import Loading from './Loading';
+import ProgressBar from './ProgressBar';
+
+const ExpiredNotice = () => {
```

Comment:
Maybe ExpiredNotice and ShowCounter could be pulled out into separate components?